src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1029,23 +1028,39 @@ public override Task<object> ExecuteScalarAsync(CancellationToken cancellationTo
             var behavior = CommandBehavior.SingleRow;
             if (!Parameters.HasOutputParameters)
                 behavior |= CommandBehavior.SequentialAccess;
-            using (var reader = await ExecuteDbDataReader(behavior, async, cancellationToken))
+            using (var reader = await ExecuteReaderAsync(behavior, async, cancellationToken))
                 return reader.Read() && reader.FieldCount != 0 ? reader.GetValue(0) : null;
         }
 
         #endregion Execute Scalar
 
         #region Execute Reader
 
+        /// <summary>
+        /// Executes the command text against the connection.
+        /// </summary>
+        protected override DbDataReader ExecuteDbDataReader(CommandBehavior behavior)
+            => ExecuteReaderAsync(behavior, async: false, CancellationToken.None).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Executes the command text against the connection.
+        /// </summary>
+        /// <param name="behavior">An instance of <see cref="CommandBehavior"/>.</param>
+        /// <param name="cancellationToken">A task representing the operation.</param>
+        /// <returns></returns>
+        protected override async Task<DbDataReader> ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
+            => await ExecuteReaderAsync(behavior, cancellationToken);
+
         /// <summary>
         /// Executes the CommandText against the Connection, and returns an DbDataReader.
         /// </summary>
         /// <remarks>
         /// Unlike the ADO.NET method which it replaces, this method returns a Npgsql-specific
         /// DataReader.
         /// </remarks>
-        /// <returns>A DbDataReader object.</returns>
-        public new NpgsqlDataReader ExecuteReader() => (NpgsqlDataReader) base.ExecuteReader();
+        /// <returns>A NpgsqlDataReader object.</returns>
+        public new NpgsqlDataReader ExecuteReader()
```

Comment:
Why not call our private ExecuteDataReader directly, instead of passing through base.ExecuteReader? This will remove a bit of useless virtual indirection.